akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -551,16 +552,17 @@ object ShardRegion {
           if (CoordinatedShutdown(context.system).getShutdownReason().isPresent)
             "" // the region will be shutdown earlier so would be confusing to say more
           else
-            s"Waiting additional [${handoffTimeout.toCoarsest}] before stopping the remaining entities.")
+            s"Waiting additional [${(entityHandOffTimeout - StopTimeoutWarningAfter).toCoarsest}] before stopping the remaining entities.")
```

Comment:
I'd expect this line to be the only change, why are the other changes needed?

Reply:
You're right, others are not needed in terms of logging.
It's just seemed cleaner 2 have all the logic on timeouts in one place - thus, entityHandOffTimeout calculation is moved inside.
And in case when entityHandOffTimeout is less than StopTimeoutWarningAfter, StopTimeoutWarningAfter timer has no further sense - thus, timers.clean

Do you want me to make a single line change and remove others ?

Reply:
Ok, fair enough, let's keep the other changes as well.